src/detect-template-buffer.c

```diff
@@ -48,6 +48,9 @@ static int DetectTemplateBufferSetup(DetectEngineCtx *de_ctx, Signature *s,
 {
     s->list = DETECT_SM_LIST_TEMPLATE_BUFFER_MATCH;
     s->alproto = ALPROTO_TEMPLATE;
+    s->flags |= SIG_FLAG_STATE_MATCH;
```

Comment:
I think if we do this, it should be done globally, so for the other keywords as well.